engine.js

```diff
@@ -122,22 +122,21 @@ async function hideThoughts() {
     const characterSettings = getCurrentCharacterSettings();
 
     const isMindReaderCharacter = Boolean(characterSettings && characterSettings.is_mind_reader);
-    const hasAccessToThought = (chatCharacter) => isMindReaderCharacter || chatCharacter.name === currentCharacter.name;
+    const hasAccessToThought = (chatThoughtName) => isMindReaderCharacter || chatThoughtName === currentCharacter.name;
 
     let promises = [];
     const lastMessageIndex = context.chat.length - 1;
     for (let i = lastMessageIndex, thoughtsCount = []; lastMessageIndex - i < settings.max_hiding_thoughts_lookup; i--) {
         if (Boolean(context.chat[i]?.is_thoughts)) {
-            const chatCharacter = context.chat[i];
-            thoughtsCount[chatCharacter.name] ??= 0;
-
-            if (thoughtsCount[chatCharacter.name] < maxThoughts && hasAccessToThought(chatCharacter)) {
+            const chatThoughtName = context.chat[i].thoughts_for || context.chat[i].name;
```

Comment:
```suggestion
            const chatThoughtName = context.chat[i]?.thoughts_for || context.chat[i].name;
```

Reply:
That doesn't do anything? The question mark after the index reference is to prevent errors if the index doesn't exist, but it already has to exist in order to be in this block.

Reply:
You are right again. I googled it, looks like I misunderstood how optional chaining works in JavaScript a little 😅